cmd/gardener-extension-provider-aws/app/migrations.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/go-logr/logr"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var nameRegex = regexp.MustCompile("extensions.gardener.cloud:provider-aws:shoot--.*:machine-controller-manager")
+
+// TODO (georgibaltiev): Remove after the release of version 1.62.0
+func purgeMachineControllerManagerRBACResources(ctx context.Context, c client.Client, log logr.Logger) error {
+	log.Info("Starting the removal of obsolete ClusterRoles and ClusterRoleBindings")
```

Comment:
Nit: For consistency reasons between logs (atm use "remove") and error messages (atm use "delete").

```suggestion
	log.Info("Starting the deletion of obsolete ClusterRoles and ClusterRoleBindings")
```